crates/bevy_input/src/gamepad.rs

```diff
@@ -78,6 +78,25 @@ impl Gamepad {
     }
 }
 
+/// Metadata associated with a `Gamepad`.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+pub struct GamepadInfo {
+    name: String,
```

Comment:
IMO this isn't high risk; we can just make the field pub and simplify the API dramatically.